plugins/modules/azure_rm_monitordatacollectionrules.py

```diff
@@ -0,0 +1,914 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: azure_rm_monitordatacollectionrules
+version_added: "3.4.0"
+short_description: Create, update and delete Data Collection Rules
+description:
+    - Create, update and delete Data Collection Rules
+
+options:
+    name:
+        description:
+            - The name of the data collection rule you're creating/changing
+        required: true
+        type: str
+    resource_group:
+        description:
+            - The name of the resource group
+        required: true
+        type: str
+    location:
+        description:
+            - Location of the data colelction rule
+            - defaults to location of exiting data collection rule or
+            - location of the resource group if unspecified
+        required: false
+        type: str
+    kind:
+        description:
+            - Kind of the data collection rule
+            - Use C(Linux) for Linux.
+            - Use C(Windows) for Windows.
+        type: str
+        choices:
+            - Linux
+            - Windows
+    description:
+        description:
+            - Description for the data collection rule
+        type: str
+    data_collection_endpoint_id:
+        description:
+            - The resource ID of the data collection endpoint that this rule can be used with
+        type: str
+    stream_declarations:
+        description:
+            - Declaration of a custom stream. Sub dict is a list of columns used by data in this stream.
+            - U(https://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2022_06_01.models.streamdeclaration?view=azure-python)
+        type: dict
+        suboptions:
+            columns:
+                description:
+                    - Declaration of a custom stream.
+                type: list
+                elements: dict
+                suboptions:
+                    name:
+                        description:
+                            - The name of the column.
+                        type: str
+                    type:
+                        description:
+                            - The type of the column data.
+                        type: str
+                        choices:
+                            - string
+                            - int
+                            - long
+                            - real
+                            - boolean
+                            - datetime
+                            - dynamic
+    data_sources:
+        description:
+            - The specification of data sources. This property is optional and can be omitted if the rule is meant to be used via direct calls to the provisioned endpoint.
+            - U(https://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2022_06_01.models.datacollectionruledatasources?view=azure-python)
```

Comment:
The URL is too long and I have no good solution. Currently, I suggest breaking the line and doing the same for the rest! Thank you!
```suggestion
            - U(https://learn.microsoft.com/en-us/python/api/azure-mgmt-monitor/azure.mgmt.monitor.v2022_06_01.
                  models.datacollectionruledatasources?view=azure-python)
```